spec/posts_controller_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
+  describe 'GET /index' do
+    before(:example) { get '/users/1/posts' }
+    it 'responds with code 200' do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'renders index view' do
+      expect(response).to render_template(:index)
+    end
+
+    it 'renders correct content' do
+      expect(response.body).to include('list of posts for a given user')
+    end
+  end
+
+  describe 'GET /show' do
+    before(:example) { get '/users/1/posts/2' }
+    it 'responds with code 200' do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'renders show view' do
+      expect(response).to render_template(:show)
```

Comment:
- [ ] You have done a good job by adding controller specs. unfortunately, this project requires that we test our models. From the project name `Blog App -validations and model specs.` we can tell what we are required to do. Let's start with adding validations to our models, and thereafter test if our model validations are working(model specs).